Added `override_container` context manager.

mauritsvanrees opened the pull request:
Used this in tests to make them pass when the standard permissive security assertions for strings has been changed.

Related to PloneHotfix20170117.

This pull request is needed to make https://github.com/zopefoundation/Zope/pull/88 pass.

leorochael commented on src/AccessControl/SimpleObjectPolicies.py:
```diff
@@ -119,3 +120,13 @@ def allow_type(Type, allowed=1):
+    ContainerAssertions[type_] = assertions
+    yield
+    if orig_container is not None:
+        ContainerAssertions[type_] = orig_container
```
If `orig_container` is `None`, shouldn't `ContainerAssertions[type_]` be deleted?

mauritsvanrees replied:
Good point. I have fixed that and have added tests for this context manager.

leorochael commented:
A few pure whitespace changes int this commit, but other than the comment above, looks ok to me...

mauritsvanrees commented:
I have addressed the comment in a new commit.

About the white space changes: I have configured http://editorconfig.org in my editor and have set it up to remove spaces at the end of the line. If it really bothers you I can of course undo that part.

leorochael commented:
It doesn't bother me much. It's just traditional in our community to do whitespace cleanup in a separate commit. I won't block this PR just for that.

davisagli commented:
FWIW, I don't mind mixing in whitespace changes so much since I learned that you can add ?w=1 to the pull request URL to see a diff ignoring whitespace differences
